spec/lib/video_info/providers/youtube_api_spec.rb

```diff
@@ -0,0 +1,392 @@
+    end
+
+    describe '#date' do
+      subject { super().date }
+      it { is_expected.to eq Time.parse('Sat Apr 12 22:25:35 UTC 2008', Time.now.utc) }
+    end
+
+    describe '#thumbnail_small' do
+      subject { super().thumbnail_small }
+      it { is_expected.to eq 'https://i.ytimg.com/vi/mZqGqE0D0n4/default.jpg' }
+    end
+
+    describe '#thumbnail_medium' do
+      subject { super().thumbnail_medium }
+      it { is_expected.to eq 'https://i.ytimg.com/vi/mZqGqE0D0n4/mqdefault.jpg' }
+    end
+
+    describe '#thumbnail_large' do
+      subject { super().thumbnail_large }
+      it { is_expected.to eq 'https://i.ytimg.com/vi/mZqGqE0D0n4/hqdefault.jpg' }
+    end
+
+    describe '#view_count' do
+      subject { super().view_count }
+      it { is_expected.to be > 4000 }
+    end
+  end
+
+  context "with video oQ49W_xKzKA", :vcr do
+    subject { VideoInfo.new('http://www.youtube.com/watch?v=oQ49W_xKzKA') }
+
+    it { expect(subject.embed_code(url_attributes: { autoplay: 1 })).to match(/autoplay=1/) }
+  end
+
+  context "with video oQ49W_xKzKA", :vcr do
+    subject { VideoInfo.new('http://www.youtube.com/watch?v=oQ49W_xKzKA') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq 'YouTube' }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'oQ49W_xKzKA' }
+    end
+  end
+
+  context "with video Xp6CXF-Cesg", :vcr do
+    subject { VideoInfo.new('http://www.youtube.com/watch?v=Xp6CXF-Cesg') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq 'YouTube' }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'Xp6CXF-Cesg' }
+    end
+  end
+
+  context "with video VeasFckfMHY in user url", :vcr do
+    subject { VideoInfo.new('http://www.youtube.com/user/EducatorVids3?v=VeasFckfMHY') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq 'YouTube' }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'VeasFckfMHY' }
+    end
+
+    describe '#url' do
+      subject { super().url }
+      it { is_expected.to eq 'http://www.youtube.com/user/EducatorVids3?v=VeasFckfMHY' }
+    end
+  end
+
+  context "with video VeasFckfMHY after params", :vcr do
+    subject { VideoInfo.new('http://www.youtube.com/watch?feature=player_profilepage&v=VeasFckfMHY') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq 'YouTube' }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'VeasFckfMHY' }
+    end
+
+    describe '#url' do
+      subject { super().url }
+      it { is_expected.to eq 'http://www.youtube.com/watch?feature=player_profilepage&v=VeasFckfMHY' }
+    end
+  end
+
+  context "with video VeasFckfMHY in path", :vcr do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.